addons/mail/static/src/models/user/user.js

```diff
@@ -195,25 +184,46 @@ function factory(dependencies) {
         }
 
         /**
-         * @override
+         * @private
+         * @returns {string}
          */
-        _updateAfter(previous) {
-            if (this.partnerDisplayName && this.partner) {
-                this.partner.update({ display_name: this.partnerDisplayName });
-            }
+        _computeDisplayName() {
+            return this.display_name || this.partner && this.partner.display_name;
         }
 
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeNameOrDisplayName() {
+            return this.display_name;
+        }
     }
 
     User.fields = {
         id: attr(),
+        display_name: attr({
+            compute: '_computeDisplayName',
+            dependencies: [
+                'partner',
+                'partnerDisplayName',
```

Comment:
`display_name` should be in, but not `partner` (it's overkill with `partnerDisplayName`, if we change partner but they both have the same display name, method shouldn't recompute).